projects/angular-auth-oidc-client/src/lib/oidc.security.service.spec.ts

```diff
@@ -631,4 +631,144 @@ describe('OidcSecurityService', () => {
       });
     });
   });
+
+  describe('isLoading$', () => {
+    it(
+      'should emit true',
+      waitForAsync(() => {
+        oidcSecurityService.isLoading$.subscribe((x) => {
+          expect(x).toBeTrue();
+        });
+      })
+    );
+
+    it(
+      'should emit false after checkauth is called',
+      waitForAsync(() => {
+        const config = { configId: 'configId1' };
+        spyOn(configurationService, 'getOpenIDConfigurations').and.returnValue(of({ allConfigs: [config], currentConfig: config }));
+        spyOn(checkAuthService, 'checkAuth').and.returnValue(of(null));
+
+        oidcSecurityService
+          .checkAuth()
+          .pipe(
+            switchMap(() =>
+              oidcSecurityService.isLoading$.pipe(
+                tap((x) => {
+                  expect(x).toBeFalse();
+                })
+              )
+            )
+          )
+          .subscribe();
+      })
+    );
+
+    it(
+      'should emit false on error in checkauth',
+      waitForAsync(() => {
+        const config = { configId: 'configId1' };
+        spyOn(configurationService, 'getOpenIDConfigurations').and.returnValue(of({ allConfigs: [config], currentConfig: config }));
+        spyOn(checkAuthService, 'checkAuth').and.returnValue(throwError(() => new Error('Error')));
+        oidcSecurityService
+          .checkAuth()
+          .pipe(
+            catchError(() =>
+              oidcSecurityService.isLoading$.pipe(
+                tap((x) => {
```

Comment:
We do not need the tap because catch error returns the isLoading, and we can check in the subscribe for the result (complete test)